opslevel/resource_opslevel_team_contact.go

```diff
@@ -0,0 +1,152 @@
+package opslevel
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/opslevel/opslevel-go"
+)
+
+func resourceTeamContact() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a team contact",
+		Create:      wrap(resourceTeamContactCreate),
+		Read:        wrap(resourceTeamContactRead),
+		Update:      wrap(resourceTeamContactUpdate),
+		Delete:      wrap(resourceTeamContactDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"last_updated": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"team": {
+				Type:        schema.TypeString,
+				Description: "The id or alias of the team to add the contact to.",
+				ForceNew:    true,
+				Required:    true,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Description:  "The type of contact",
+				ForceNew:     false,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(opslevel.AllContactType(), false),
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The display name of the contact.",
+				ForceNew:    false,
+				Required:    true,
+			},
+			"value": {
+				Type:        schema.TypeString,
+				Description: "A value of the contact.",
+				ForceNew:    false,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceTeamContactCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	input := opslevel.ContactInput{
+		Type:        opslevel.ContactType(d.Get("type").(string)),
+		DisplayName: d.Get("name").(string),
+		Address:     d.Get("value").(string),
+	}
+	resource, err := client.AddContact(d.Get("team").(string), input)
+	if err != nil {
+		return err
+	}
+	d.SetId(resource.Id.(string))
+
+	return resourceTeamContactRead(d, client)
+}
+
+func resourceTeamContactRead(d *schema.ResourceData, client *opslevel.Client) error {
+	id := d.Id()
+
+	// Handle Import by spliting the ID into the 2 parts
+	parts := strings.SplitN(id, ":", 2)
+	if len(parts) == 2 {
+		d.Set("team", parts[0])
+		id = parts[1]
+		d.SetId(id)
+	}
+
+	identifier := d.Get("team").(string)
+	var err error
+	var team *opslevel.Team
+	if opslevel.IsID(identifier) {
```

Comment:
I really hate this - we should probably push this complexity down into the opslevel-go library.  Not sure if now is the best time to do that though.

Reply:
Ah yeah, that would be nice... I'm doing this in a few places now too.  Would it just involve reducing `GetTeam` and `GetTeamWithAlias` down to 1 function with this check in there?

Reply:
Ya - we would need to adjust all our functions that take "graphql.ID" and instead take "IdentifierInput" and then make a different mutation weather its an Alias or an ID.  Its a pretty massive change so nows likely not the time.